### PR TITLE
Add UEFI image IBBSegment adding

### PR DESCRIPTION
**Pull request by ChriMarMe**
Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: pkg/provisioning/cbnt/tools.go [hunk 1]
```diff
@@ -332,3 +332,27 @@ func FindAdditionalIBBs(imagepath string) ([]bootpolicy.IBBSegment, error) {
...
+	if err != nil {
+		return ibbs, err
+	}
+	fitentries, err := fit.GetEntries(img)
+	if err != nil {
+		return ibbs, err
+	}
+	for _, entry := range fitentries {
+		switch entry.GetType() {
+		case fit.EntryTypeBIOSStartupModuleEntry:
```
Comment (ArthurHeymans) [inline pkg/provisioning/cbnt/tools.go:350]: if entry.GetType() == fit.EntryTypeBIOSStartupModuleEntry {}

File: cmd/cbnt-prov/cmd.go [hunk 2]
```diff
@@ -129,6 +129,7 @@ type generateBPMCmd struct {
 	IbbSegsize  uint32               `flag optional name:"ibbsegsize" help:"Value for IBB segment structure"`
 	IbbSegFlag  uint16               `flag optional name:"ibbsegflag" help:"Reducted"`
 	Coreboot    bool                 `flag optional name:"coreboot" help:"Required when BIOS binary file is a coreboot image"`
+	Uefi        bool                 `flag optional name:"uefi" help:"Required when BIOS binary file is a UEFI image"`
```
Comment (ArthurHeymans) [inline cmd/cbnt-prov/cmd.go:132]: Should the kong 'xox:"coreboot"' flag be used to make them mutually exclusive?
Coreboot will not be adding fit type 7 I suppose and it avoids confusion.

File: cmd/cbnt-prov/cmd.go [hunk 3]
```diff
@@ -430,14 +431,22 @@ func (g *generateBPMCmd) Run(ctx *context) error {
 			seg.Flags = g.IbbSegFlag
 			se.IBBSegments = append(se.IBBSegments, seg)
 		}
-		if g.Coreboot {
+		if g.Coreboot && !g.Uefi {
```
Comment (ArthurHeymans) [inline cmd/cbnt-prov/cmd.go:434]: With xor in the cmdline flags, can we just do 'if g.Coreboot {' ?

Comment (ChriMarMe) [reply cmd/cbnt-prov/cmd.go]: true

**Review by ArthurHeymans — DISMISSED** [review]

File: cmd/cbnt-prov/cmd.go [hunk 4]
```diff
@@ -130,7 +130,8 @@ type generateBPMCmd struct {
 	IbbSegbase  uint32               `flag optional name:"ibbsegbase" help:"Value for IbbSegment structure"`
 	IbbSegsize  uint32               `flag optional name:"ibbsegsize" help:"Value for IBB segment structure"`
 	IbbSegFlag  uint16               `flag optional name:"ibbsegflag" help:"Reducted"`
-	Coreboot    bool                 `flag optional name:"coreboot" help:"Required when BIOS binary file is a coreboot image"`
+	Coreboot    bool                 `flag optional xor:"uefi,coreboot" name:"coreboot" help:"Required when BIOS binary file is a coreboot image"`
```
Comment (zaolin) [inline cmd/cbnt-prov/cmd.go:133]: why not auto-detect???

**Review by ArthurHeymans — DISMISSED** [review]

File: pkg/provisioning/cbnt/tools.go [hunk 5]
```diff
@@ -289,10 +289,29 @@ func FindAdditionalIBBs(imagepath string) ([]bootpolicy.IBBSegment, error) {
...
+				ibb := bootpolicy.NewIBBSegment()
+				ibb.Base = uint32(entry.GetHeaders().Address.Pointer())
+				ibb.Size = entry.GetHeaders().Size.Uint32() << 4
+				ibbs = append(ibbs, *ibb)
+			}
+		}
+		return ibbs, nil
 	}
-
+	// From here we consider it is a coreboot image
```
Comment (zaolin) [inline pkg/provisioning/cbnt/tools.go:314]: better check for FMAP in BIOS region. fiano should support that ccheck

Comment (ChriMarMe) [reply pkg/provisioning/cbnt/tools.go:314]: cbfs.NewImage(...) checks for FMAP named COREBOOT. Why is that not sufficient?

Comment (ArthurHeymans) [reply pkg/provisioning/cbnt/tools.go:314]: That's sufficient and even desirable for buildtesting in coreboot where no IFD is present.

**Review by ArthurHeymans — APPROVED** [review]

**Pull request merged**